### PR TITLE
Powermax e2e fixes

### PR DESCRIPTION
**Pull request by kumarkgosa**
# Description
Updating Powermax test files to fix e2e 

**Note:** Missing controller permissions for namespace. Added to match helm-chart permissions.
 - Previous error: 
 ```
2024/07/22 18:36:32 getClusterUuid error:  namespaces "kube-system" is forbidden: User "system:serviceaccount:test-vxflexos:test-vxflexos-controller" cannot get resource "namespaces" in API group "" in the namespace "kube-system"
 ```
 - Fixed:
 ```
2024/07/22 21:43:33 getClusterUuid got uuid:  2500dee2-80f8-4322-ab59-917142133869
 ```

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1221 ||
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Run the E2E scenario of Installing Powermax with Authorization and Reverse Proxy enabled. Ran the Cert-CSI test. 

image

### CODE REVIEW

File: operatorconfig/driverconfig/powermax/v2.11.0/node.yaml [hunk 1]
```diff
@@ -127,7 +127,7 @@ spec:
               value: "<X_CSI_POWERMAX_ISCSI_ENABLE_CHAP>"
             - name: X_CSI_POWERMAX_PROXY_SERVICE_NAME
               value: "csipowermax-reverseproxy"
-            - name: X_CSI_NODE_CHROOT
+            - name: X_CSI_ISCSI_CHROOT
```
Comment (delldubey) [inline operatorconfig/driverconfig/powermax/v2.11.0/node.yaml:130]: Please revert the change, as the new driver 2.11.0 look out for X_CSI_NODE_CHROOT.

**Review by delldubey — CHANGES_REQUESTED** [review]

File: tests/e2e/testfiles/scenarios.yaml [hunk 2]
```diff
@@ -1,69 +1,53 @@
-# List of E2E Tests Scenarios
-
-- scenario: "Install Authorization Proxy Server V1"
+#List of E2E Tests Scenarios
+- scenario: "Install Authorization Proxy Server"
   paths:
-    - "testfiles/authorization-templates/csm_authorization_proxy_server_n_minus_1.yaml"
+    - "testfiles/authorization-templates/csm_authorization_proxy_server.yaml"
```
Comment (harshitap26) [inline tests/e2e/testfiles/scenarios.yaml:4]: We should continue to use n-1 YAML as testfiles/authorization-templates/csm_authorization_proxy_server.yaml" is used for Stateless Authorization deployment. We need to have separate scenarios for V1 and V2.

**Review by harshitap26 — CHANGES_REQUESTED** [review]

File: tests/e2e/testfiles/scenarios.yaml [hunk 3]
```diff
@@ -1564,44 +1570,52 @@
...
-    - "Validate custom resource [1]"
-    - "Validate [powermax] driver from CR [1] is installed"
-    - "Validate [observability] module from CR [1] is not installed"
-    - "Enable [observability] module from CR [1]"
-    - "Validate [powermax] driver from CR [1] is installed"
-    - "Validate [observability] module from CR [1] is installed"
-    - "Disable [observability] module from CR [1]"
-    - "Validate [powermax] driver from CR [1] is installed"
-    - "Validate [observability] module from CR [1] is not installed"
+    - "Install Authorization CRDs [1]"
```
Comment (harshitap26) [inline tests/e2e/testfiles/scenarios.yaml:1596]: Not required for Powermax as stateless auth is not supported for Powermax.

File: tests/e2e/testfiles/scenarios.yaml [hunk 4]
```diff
@@ -1,69 +1,53 @@
-# List of E2E Tests Scenarios
-
-- scenario: "Install Authorization Proxy Server V1"
+#List of E2E Tests Scenarios
+- scenario: "Install Authorization Proxy Server"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/scenarios.yaml:2]: The authorization scenarios were updated recently in PR #641. Please maintain those changes to not introduce breaking changes to the authorization scenarios. Thank you.

**Review by shaynafinocchiaro — CHANGES_REQUESTED** [review]

File: tests/e2e/testfiles/scenarios.yaml [hunk 5]
```diff
@@ -1463,9 +1463,8 @@
     # Last steps perform Clean UP
     - "Restore template [testfiles/powerflex-templates/powerflex-secret-template.yaml] for [pflex]"
     - "Restore template [testfiles/powerflex-templates/powerflex-storageclass-template.yaml] for [pflex]"
-    - "Restore template [testfiles/application-mobility-templates/csm_application_mobility_no_velero.yaml] for [application-mobility]"
+    - "Restore template [testfiles/application-mobility-templates/csm_application_mobility_with_pflex.yaml] for [application-mobility]"
```
Comment (kumarp20) [inline tests/e2e/testfiles/scenarios.yaml:1466]: Are these changes intentional and needed in this PR ? They seem unrelated to Powermax e2e tests.

File: tests/e2e/testfiles/storage_csm_powermax.yaml [hunk 6]
```diff
@@ -156,53 +177,106 @@ spec:
...
-      configVersion: v1.10.0
+      configVersion: v1.9.1
       components:
         - name: podmon-controller
-          image: dellemc/podmon:v1.10.0
+          image: dellemc/podmon:v1.9.1
           imagePullPolicy: IfNotPresent
           args:
-            - "--labelvalue=csi-powermax"
+            - "--labelvalue=csi-isilon"
```
Comment (kumarp20) [inline tests/e2e/testfiles/storage_csm_powermax.yaml:279]: why these changes ?

**Review by kumarp20 — COMMENTED** [review]
Is this PR intended to add E2E tests for KRV-22771 ? The only change that went into KRV-22771 is that in Operator install of powermax driver, there is now an option to deploy the reverse-proxy as a sidecar, and this option is to be used when powermax driver has auth enabled. 
So the E2E tests that can be added are:
1. Install powermax driver with reverse-proxy Deploy as SideCar option set to false, run cert-csi to validate that the driver works properly. (new scenario with new test files): This can be covered by a scenario similar to https://github.com/dell/csm-operator/blob/main/tests/e2e/testfiles/scenarios.yaml#L1555, but with DeployAsSidecar set to false.
2. Install powermax driver with reverse -proxy deploy as SideCar set to true; run cert-csi to validate that the driver works properly. (new scenario with new test files) . This is covered by scenario here, as per Utkarsh: https://github.com/dell/csm-operator/blob/main/tests/e2e/testfiles/scenarios.yaml#L1555
3. Install and Configure Authoriztaion sever. Install powermax dirver with auth enabled and reverse-proxy deployed as sidecar; run cert-csi to validate that the driver works properly. (new scenario with new test files which Harshita has been working on)

No other changes seem relevant to KRV-22771. 
@kumarkgosa @harshitap26 Can we remove all unnecessary changes from this PR. Any inadvertent changes may break other E2e tests, so please be mindful of what we change.

File: operatorconfig/driverconfig/powermax/v2.11.0/node.yaml (hunk 1 above)
Comment (harshitap26) [reply operatorconfig/driverconfig/powermax/v2.11.0/node.yaml]: Reverted the changes.

File: tests/e2e/testfiles/storage_csm_powermax.yaml (hunk 6 above)
Comment (kumarkgosa) [reply tests/e2e/testfiles/storage_csm_powermax.yaml]: changed. resolved

File: tests/e2e/testfiles/scenarios.yaml (hunk 2 above)
Comment (harshitap26) [reply tests/e2e/testfiles/scenarios.yaml]: Done

File: tests/e2e/testfiles/scenarios.yaml (hunk 3 above)
Comment (harshitap26) [reply tests/e2e/testfiles/scenarios.yaml]: Reverted the changes back to Auth V1

File: tests/e2e/testfiles/scenarios.yaml (hunk 5 above)
Comment (kumarkgosa) [reply tests/e2e/testfiles/scenarios.yaml]: The scenarios weren't ready while creating the draft. changed now.

**Review by harshitap26 — APPROVED** [review]

**Review by atye — APPROVED** [review]

**Review by harshitap26 — APPROVED** [review]

File: tests/e2e/testfiles/powermax-templates/powermax_reverse_proxy_config.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# Copyright © 2023 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (gallacher) [inline tests/e2e/testfiles/powermax-templates/powermax_reverse_proxy_config.yaml:1]: 2024

File: tests/e2e/testfiles/scenarios.yaml [hunk 8]
```diff
@@ -299,7 +299,7 @@
   paths:
     - "testfiles/storage_csm_powerscale_replica.yaml"
   tags:
-    # - "powerscale"
+    - "powerscale"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/scenarios.yaml:302]: Some tags got uncommented in this file, should they stay commented?

Comment (kumarkgosa) [reply tests/e2e/testfiles/scenarios.yaml:302]: Its good, we got it tested.

File: tests/e2e/testfiles/powermax-templates/powermax_reverse_proxy_config.yaml (hunk 7 above)
Comment (kumarkgosa) [reply tests/e2e/testfiles/powermax-templates/powermax_reverse_proxy_config.yaml:1]: Okay changing...

**Review by shaynafinocchiaro — APPROVED** [review]

**Review by atye — APPROVED** [review]
